packages/core/src/defer/triggering.ts

```diff
@@ -74,16 +74,17 @@ export function scheduleDelayedTrigger(
   const tNode = getCurrentTNode()!;
   const injector = lView[INJECTOR]!;
   const lDetails = getLDeferBlockDetails(lView, tNode);
-  const tDetails = getTDeferBlockDetails(lView[TVIEW], tNode);
+  const isServerRendered =
+    lDetails[SSR_BLOCK_STATE] !== null && lDetails[SSR_BLOCK_STATE] === DeferBlockState.Complete;
```

Comment:
I guess the intention was to check for the `SSR_UNIQUE_ID` field value presence?
```suggestion
  const isServerRendered =
    lDetails[SSR_UNIQUE_ID] !== null && lDetails[SSR_BLOCK_STATE] === DeferBlockState.Complete;
```

Reply:
I also think that there should be no conditions in this function and we should move the check to an instruction level (i.e. to a call site).